### PR TITLE
Update API convert

### PR DESCRIPTION
**Pull request by ROckDog22**
Update API convert

### CODE REVIEW

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+## [ 组合替代实现 ] torch.Tensor.index_copy_
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md]: 看起来是 torch参数更多

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#### 示例 2: 索引的维度不为 0
+```python
+# torch 写法
+y = x.index_copy_(2, index, t)
+
+# paddle 写法
+dim = list(x.shape)
+for i0 in range(dim[0]):
+    for i1 in range(dim[1]):
+        x[i0, i1, :] = x[i0][i1].scatter_(index, t[i0][i1])
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md]: 切片采取x[i0, i1] 的形式吧，不用切两次性能变低

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+# torch 写法
+x.index_copy_(0, index, t)
+
+# paddle 写法
+x.scatter_(index, t)
+```
+#### 示例 2: 索引的维度不为 0
+```python
+# torch 写法
+y = x.index_copy_(2, index, t)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md]: 这个y=不需要，写最简单的示例就可以。避免非必要代码增加理解成本

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+# paddle 写法
+dim = list(x.shape)
+for i0 in range(dim[0]):
+    for i1 in range(dim[1]):
+        x[i0, i1, :] = x[i0][i1].scatter_(index, t[i0][i1])
+y = x.clone()
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/tensor.Torch.index_copy_.md]: 非必要代码

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.requires_grad.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+##  [ 无参数 ] torch.Tensor.requires_grad
+
+### [torch.Tensor.requires_grad](https://pytorch.org/docs/2.0/generated/torch.Tensor.requires_grad.html#torch.Tensor.requires_grad)
+
+```python
+torch.Tensor.requires_grad
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.requires_grad.md:6]: 这两个API是不一致的，写一个转写示例吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.to.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+## torch.Tensor.to
+### [torch.Tensor.to](https://pytorch.org/docs/2.0/generated/torch.Tensor.to.html#torch-tensor-to)
+
+```python
+torch.Tensor.to(dtype, non_blocking=False, copy=False, memory_format=torch.preserve_format)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/Tensor/torch.Tensor.to.md:9]: 这个torch.Tensor.to比较复杂，你仔细看下他所有的用法，
https://pytorch.org/docs/stable/generated/torch.Tensor.to.html?highlight=torch+tensor#torch.Tensor.to

infoflow 2023-04-04 12-06-29

可能得分成多个来写，参考torch.max

File: docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
+## [ 组合替代实现 ] torch.cdist
+
+### [torch.dist](https://pytorch.org/docs/stable/generated/torch.cdist.html?highlight=cdist#torch-cdist)
+
+```python
+torch.Tensor.cdist(x1, x2, p=2.0, compute_mode='use_mm_for_euclid_dist_if_necessary')
+```
+
+PaddlePaddle 目前无对应 API，可使用如下代码组合实现该 API 转写。
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md]: 可简述下功能，再说我们没有

File: docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+### 转写示例
+```python
+# torch 写法
+a = torch.tensor([[0.9041,  0.0196], [-0.3108, -2.4423], [-0.4821,  1.059]])
+b = torch.tensor([[-2.1763, -0.4713], [-0.6986,  1.3702]])
+c = torch.cdist(a, b, p=2)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md]: 不用加个c=
非必要代码

File: docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        for j in range(sa[1]):
+            for k in range(sb[0]):
+                x[i, j, k] = paddle.dist(a[i][j], b[k], p=2)
+else:
+    x = paddle.empty(shape=(sa[0], sa[1], sb[1]), dtype='float32')
+    for i in range(sa[0]):
+        for j in range(sa[1]):
+            for k in range(sb[1]):
+                x[i, j, k] = paddle.dist(a[i, j], a[i, k], p=2)
+c = x.clone()
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md]: 去除非必要代码

File: docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+```
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/functional/torch.cdist.md:29]: 这些组合实现需要确保正确哈，否则引入问题就很麻烦

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+                      data_format='NCDHW',
+                      name=None)
+```
+
+两者功能一致但参数不一致，部分参数名不同，具体如下：
+### 参数映射
+| PyTorch       | PaddlePaddle | 备注                                                   |
+| ------------- | ------------ | ------------------------------------------------------ |
+| num_features           | num_features      | 表示输入 Tensor 通道数。                                     |
+| eps           | epsilon      | 为了数值稳定加在分母上的值。                                     |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md]: 仅参数名不同。需要写一下

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+| momentum           | momentum      | 表示归一化函数中的超参数， PyTorch 和 Paddle 公式实现细节不一致，两者正好是相反的，需要进行转写。                                     |
+| -             | weight_attr  | 指定权重参数属性的对象。如果为 False, 则表示每个通道的伸缩固定为 1，不可改变。默认值为 None，表示使用默认的权重参数属性。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md]: Paddle多的参数需要说明如何设置

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+| -             | bias_attr    | 指定偏置参数属性的对象。如果为 False, 则表示每一个通道的偏移固定为 0，不可改变。默认值为 None，表示使用默认的偏置参数属性。 |
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md]: Paddle多的参数需要说明如何设置

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+| -             | data_format  | 指定输入数据格式， Pytorch 无此参数，Paddle 保持默认即可。 |
+| affine        | -            | 是否进行反射变换， PaddlePaddle 无此参数，需要进行转写。         |
+| track_running_stats | - | 无对应的参数，目前设置 track_running_stats 和 momentum 是无效的。之后的版本会修复此问题。|
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md]: 是否转写需要写明白

File: docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+| device | - | Tensor 的设备，一般对网络训练结果影响不大，可直接删除。         |
+| dtype| - | 指定权重参数属性的对象，一般对网络训练结果影响不大，可直接删除。 |
+### 转写示例
+#### affine：是否进行反射变换
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/nn/torch.nn.InstanceNorm3d.md]: momentum如何转写？

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Generator.md [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+### 转写示例
+```python
+# torch 写法
+g_cpu = torch.Generator()
+g_cuda = torch.Generator(device='cuda')
+
+# paddle 写法
+g_cpu = paddle.fluid.core.default_cpu_generator()
+device = paddle.device.get_device()
+g_cuda = paddle.fluid.core.default_cuda_generator(int(device[-1]))
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Generator.md:20]: 这个有int吗

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Size.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
+## [ 组合替代实现 ] torch.Generator
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Size.md]: 标头写错了

File: docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Size.md [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+
+PaddlePaddle 目前无对应 API，可使用如下代码组合实现该 API 转写。
+
+### 转写示例
+```python
+# torch 写法
+shape = torch.Size((1,2,3,4))
+
+# paddle 写法
+shape = paddle.shape(paddle.empty((1, 2, 3, 4)))
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/others/torch.Size.md]: torch这个返回类型torch.Size其实相当于一个list，不用转写成paddle.Tensor，我们就用Tensor.shape替代，返回是一个list一样的

File: docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.data.BatchSampler.md [hunk 19]
```diff
@@ -31,9 +31,11 @@ paddle.io.BatchSampler(dataset=None,
...
 # 若 sampler 为 Iterable 数据类型，则需要按如下方式转写
 # Pytorch 写法
-torch.utils.data.BatchSampler(sampler=[1., 2., 3., 4.], 3, False)
+torch.utils.data.BatchSampler(sampler=[1., 2., 3., 4.], batch_size=3, drop_last = False)
 
 # Paddle 写法
-sampler = paddle.io.Sampler([1., 2., 3., 4.])
-paddle.io.BatchSampler(sampler, 3, False)
+sampler = [1.0, 2.0, 3.0, 4.0]
+sampler = sampler if issubclass(sampler.__class__, paddle.fluid.dataloader.
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/utils/torch.utils.data.BatchSampler.md]: 是不是能用isinstance这种写法，这个写法可能用的比较少
